### ISSUE TITLE
Credentials via HashiCorp Vault Secret Lookup does not work with non-official certificates

### ISSUE THREAD

**Issue by tlx1** [ansible/awx#4274]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
If you use a certificate signed by non-official CA (or self-signed) for your HashiCorp Vault installation, you cannot use the credential plugin "HashiCorp Vault Secret Lookup".
Adding the certificate to /etc/pki/ca-trust/ does not work, it still gets ignored.

##### ENVIRONMENT
* AWX version: 5.0.0
* AWX install method: kubernetes (official image from dockerhub)
* Ansible version: 2.8.1
* Operating System: Fedora
* Web Browser: Firefox

##### STEPS TO REPRODUCE

Set up a vault server with a self-signed (or signed by non-official CA) and try to use the credentials plugin.

##### EXPECTED RESULTS

Being able to use the credentials plugin to lookup secrets in vault.

##### ACTUAL RESULTS

The following error appears:

```
HTTPSConnectionPool(host='vault.internal', port=8200): Max retries exceeded with url: /v1/secret/my-secret (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'tls_process_server_certificate', 'certificate verify failed')],)",),))
```

##### ADDITIONAL INFORMATION

Could probably be fixed, similar to the credential plugin 'awx/awx/main/credential_plugins/conjur.py', where you can add a CA certificate to validate against.

**Comment by ryanpetrello** [comment]
@tlx1 yep, it looks to me like the Vault plugin doesn't have support for a `cacert` argument.  You have any interest in trying it out and opening a PR?

**Comment by wenottingham** [comment]
You can in theory use REQUESTS_CA_BUNDLE to point to an alternate cert bundle. You'll likely need to set this env in the process that executes AWX (so in the container) - this isn't via AWX_TASK_ENV.

**Comment by olia-dev** [comment]
Added a pull request that should fix this issue.

**Comment by one-t** [comment]
Verified that this has been fixed in 3.6

Steps to test:
1. Start vault in a container
2. generate a self-signed certificate
3. run nginx in another container as a proxy for vault, using this cert
4. create a credential lookup in awx
5. use the cert.pem in the CA field
6. test secret lookup
